core/src/com/protoevo/utils/GLComputeShaderRunner.java

```diff
@@ -124,16 +172,16 @@ public byte[] processImage(byte[] pixels, byte[] result, int w, int h) {
         return processImage(pixels, result, w, h, 4);
     }
 
+
     public byte[] processImage(byte[] pixels, byte[] result, int w, int h, int c) {
-        if (!initialized) {
-            ApplicationManager.ensureWindowUpToDate();
-            if (ApplicationManager.window == NULL) {
-                System.out.println("GLComputeShader cannot run because window is null");
-                return result;
-            } else {
-                initialise();
-            }
-        }
+        glThread.execute(() -> {
```

Comment:
What happens if the thread is already executing at this point? Or is it blocking?

Reply:
Yea it's blocking, the part that makes it so is the futureTask.get() : https://github.com/DylanCope/ProtoEvo/pull/3/files#diff-beeefb6e46276d407fd4189b5cc69e6a39f56f546f34626f9f11c782a2fadf29R34